lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -0,0 +1,234 @@
+        required: False
+        description: Network share user in the format 'user@domain' or 'domain\\user' if user is
+            part of a domain else 'user'. This option is mandatory for CIFS Network Share.
+    share_pwd:
+        required: False
+        description: Network share user password. This option is mandatory for CIFS Network Share.
+    share_mnt:
+        required: True
+        description: Local mount path of the network share with read-write permission for ansible user.
+            This option is mandatory for Network Share.
+    reboot:
+        required: False
+        description: Whether to reboots after applying the updates or not.
+        default: False
+        type: bool
+    job_wait:
+        required:  True
+        description: Whether to wait for job completion or not.
+        type: bool
+        default: True
+    catalog_file_name:
+        required: False
+        description: Catalog file name relative to the I(share_name).
+        type: str
+        default: 'Catalog.xml'
+
+requirements:
+    - "omsdk"
+    - "python >= 2.7.5"
+author: "Rajeev Arakkal (@rajeevarakkal)"
+"""
+
+EXAMPLES = """
+---
+- name: Update firmware from repository on a Network Share
+  dellemc_idrac_firmware:
+       idrac_ip:   "{{ idrac_ip }}"
+       idrac_user: "{{ idrac_user }}"
+       idrac_pwd:  "{{ idrac_pwd }}"
+       share_name: "xx.xx.xx.xx:/share"
+       share_user: "{{ share_user }}"
+       share_pwd:  "{{ share_pwd }}"
+       share_mnt: "/mnt/share"
+       reboot:     True
+       job_wait:   True
+       catalog_file_name:  "Catalog.xml"
+"""
+
+RETURN = """
+dest:
+    description: Updates firmware from a repository on a network share (CIFS, NFS).
+    returned: success
+    type: dict
+    sample: {
+        'CompletionTime': '2018-01-24T00:47:33',
+        'Id': 'JID_XXXXXXXXXXXX',
+        'JobState': 'Completed',
+        'JobType': 'ImportConfiguration',
+        'Message': 'Successfully imported and applied Server Configuration Profile.',
+        'MessageId': 'SYSXXX',
+        'Name': 'Import Configuration',
+        'StartTime': 'TIME_NOW',
+        'Status': 'Success',
+    }
+"""
+
+
+from ansible.module_utils.remote_management.dellemc.dellemc_idrac import iDRACConnection
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from omsdk.sdkcreds import UserCredentials
+    from omsdk.sdkfile import FileOnShare
+    HAS_OMSDK = True
+except ImportError:
+    HAS_OMSDK = False
+
+
+def _validate_catalog_file(params):
+    invalid, message = False, ""
+    if not params.get("catalog_file_name"):
+        invalid, message = True, "Invalid file name or invalid file extensions."
+    elif params.get("catalog_file_name"):
+        file_name = params.get("catalog_file_name").lower()
+        if not file_name.endswith("xml"):
+            invalid, message = True, "Invalid file name or invalid file extensions."
+    return invalid, message
+
+
+def run_update_fw_from_nw_share(idrac, module):
```

Comment:
Python is a language with encourages clear and human-readable communication. While keeping things short we don't sacrifice keystrokes. fw and nw are misleading for the reader (especially where this thing is called). firewall? northwest?
```suggestion
def update_firmware_from_netshare(idrac, module):
```

Reply:
function renamed to `update_firmware`